primitives/src/traits/zeitgeist_multi_reservable_currency.rs

```diff
@@ -44,26 +44,30 @@ where
 {
     fn accounts_by_currency_id(
         currency_id: Self::CurrencyId,
-    ) -> Result<(usize, Vec<(T::AccountId, AccountData<Self::Balance>)>), DispatchError> {
+    ) -> Result<(usize, Vec<T::AccountId>), DispatchError> {
         let mut total = 0;
         #[allow(
             // Iterator will never yield more than `usize::MAX` elements
             clippy::integer_arithmetic
         )]
         let accounts = <Accounts<T>>::iter()
-            .filter_map(|(k0, k1, v)| {
-                total += 1;
-                if k1 == currency_id { Some((k0, v)) } else { None }
+            .filter_map(|(k0, k1, _v)| {
+                if k1 == currency_id {
+                    total += 1;
+                    Some(k0)
+                } else {
+                    None
+                }
             })
             .collect();
         Ok((total, accounts))
     }
 
     fn destroy_all<I>(currency_id: Self::CurrencyId, accounts: I) -> Result<(), DispatchError>
     where
-        I: Iterator<Item = (T::AccountId, AccountData<Self::Balance>)>,
+        I: Iterator<Item = T::AccountId>,
     {
-        for (k0, _) in accounts {
+        for k0 in accounts {
             <Accounts<T>>::remove(k0, currency_id);
```

Comment:
Yes this will be used, but I don't see any other option.

Reply:
I realised, that you can throw an iterator in `destroy_all`, so that you can divide the work over multiple blocks if needed. The `TotalIssuance` is removed right at the first call of `destroy_all`. Hopefully no bad consequences with that.

So in swaps and pm-pallet the `accounts_by_currency_id` is called, in which a vector is returned at the moment. and after that the accounts are destroyed by `destroy_all`. But at some point the vector is too large. So my suggestion is, that we should return an iterator for `accounts_by_currency_id` and should delegate the work for the calls [here](https://github.com/zeitgeistpm/zeitgeist/blob/48ba93bb12cd0520c0ee0c878dc9ca0446b9454d/zrml/prediction-markets/src/lib.rs#L1695-L1701) for example to some kind of multiple block deletion. 

https://github.com/zeitgeistpm/zeitgeist/blob/48ba93bb12cd0520c0ee0c878dc9ca0446b9454d/zrml/prediction-markets/src/lib.rs#L1695-L1701

This above is a time bomb.